pkg/plugin/plugin.go

```diff
@@ -0,0 +1,42 @@
+// package plugin provides the plugin interface for generating SPDX SBOMs
+package plugin
+
+import (
+	"errors"
+
+	"github.com/spdx/tools-golang/spdx/common"
+)
+
+// Plugin is the interface a plugin must implement
+// GetSpdxDocument returns an object of type AnyDocument
+type Plugin interface {
+	GetSpdxDocument() (common.AnyDocument, error)
+}
+
+// plugins stores a mapping of the plugin name and its
+// corresponding Plugin object
+var plugins = make(map[string]Plugin)
+
+// Register registers a plugin by name
+// A plugin can use this function to create a plugin entry in their
+// init function
+func Register(name string, p Plugin) {
+	if p == nil {
+		panic("A Plugin object is required, but is nil")
```

Comment:
Since this is a plugin system potentially used by third parties, don't panic here. Return an error and let the application handle it.